api/envoy/service/status/v3/csds.proto

```diff
@@ -84,19 +84,20 @@ message ClientStatusRequest {
 }
 
 // Detailed config (per xDS) with status.
-// [#next-free-field: 7]
+// [#next-free-field: 8]
 message PerXdsConfig {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.service.status.v2.PerXdsConfig";
 
   // Config status generated by management servers. Will not be present if the
   // CSDS server is an xDS client.
-  ConfigStatus status = 1;
+  ConfigStatus status = 1 [(udpa.annotations.field_migrate).oneof_promotion = "status_config"];
 
   // Client config status is populated by xDS clients. Will not be present if
   // the CSDS server is an xDS server. No matter what the client config status
   // is, xDS clients should always dump the most recent accepted xDS config.
-  ClientConfigStatus client_status = 7;
+  ClientConfigStatus client_status = 7
```

Comment:
I think you might need to have a `oneof` existing. I suggest moving this field lexically before the existing `status` and wrapping it in a `oneof status_config`. That should allow the v4alpha generation to work.

Reply:
Lidi, I think you still need to address this comment.

Reply:
According to the style in the code base, the `oneof_promotion ` annotation seems appear in pairs ([example 1](https://github.com/envoyproxy/envoy/blob/99471fd8f4ef7406f50cf41cf1cfa22bbdfeacc7/api/envoy/extensions/transport_sockets/tls/v3/secret.proto#L36), [example 2](https://github.com/envoyproxy/envoy/blob/99471fd8f4ef7406f50cf41cf1cfa22bbdfeacc7/api/envoy/extensions/common/tap/v3/common.proto#L36)). Splitting the oneof options into different layers might breaks the consistency, and slightly weaken the semantic intention of the upcoming migration.

So, in my mind, keeping `oneof_promotion` in v3 and real oneof in v4alpha is the better way.

Reply:
Yes, that's what I meant.

The v4alpha file looks right now.  The last time I looked, I thought it was still wrong, but maybe I was looking at an old snapshot.

Anyway, this looks great.  Thanks!